### PR TITLE
Ports baystation armor system

### PR DESCRIPTION
**Pull request by Alberyk**
Ports Baystation12/Baystation12#12658

Changing how the calculation of armor works , instead of just being a check if it is protected fully, half or nothing. Making armor more reliable and less rng dependant.

Also, uranium meteors will irradiate nearby people when they explode.

### CODE REVIEW

**Comment by Lohikar** [comment]
Does this take into account that `armor` can be null in our code?

**Comment by Alberyk** [comment]
I did not run in any issues while testing it, so, sems to handle this fine.

File: code/_helpers/mobs.dm [hunk 1]
```diff
@@ -14,6 +14,9 @@
 /mob/get_mob()
 	return src
 
+//helper for inverting armor blocked values into a multiplier
+#define blocked_mult(blocked) max(1 - (blocked/100), 0)
```
Comment (Lohikar) [inline code/_helpers/mobs.dm]: Defines should go in the `__defines` folder.

File: code/game/gamemodes/meteor/meteors.dm [hunk 2]
```diff
@@ -256,7 +256,9 @@
 	meteordrop = /obj/item/weapon/ore/uranium
 
 /obj/effect/meteor/irradiated/meteor_effect()
-	new /obj/effect/decal/cleanable/greenglow(get_turf(src)) //todo: make this irradiate the place it lands
+	new /obj/effect/decal/cleanable/greenglow(get_turf(src))
+	for(var/mob/living/L in view(5, src))
+		L.apply_effect(40, IRRADIATE, blocked = L.getarmor(null, "rad"))
```
Comment (Lohikar) [inline code/game/gamemodes/meteor/meteors.dm]: \> unlogged changes

Comment (Alberyk) [inline code/game/gamemodes/meteor/meteors.dm]: It was todo flag, added to the pr description now.

File: code/__defines/mobs.dm [hunk 3]
```diff
@@ -196,3 +196,6 @@
 #define TASTE_NORMAL 1 //anything below 15%
 #define TASTE_DULL 0.5 //anything below 30%
 #define TASTE_NUMB 0.1 //anything below 150%
+
+//helper for inverting armor blocked values into a multiplier
+#define blocked_mult(blocked) max(1 - (blocked/100), 0)
```
Comment (Lohikar) [inline code/__defines/mobs.dm]: This should be capitalized as it's a define.

File: code/game/gamemodes/meteor/meteors.dm [hunk 4]
```diff
@@ -256,7 +256,9 @@
 	meteordrop = /obj/item/weapon/ore/uranium
 
 /obj/effect/meteor/irradiated/meteor_effect()
-	new /obj/effect/decal/cleanable/greenglow(get_turf(src)) //todo: make this irradiate the place it lands
+	new /obj/effect/decal/cleanable/greenglow(get_turf(src))
+	for(var/mob/living/L in view(5, src))
```
Comment (Lohikar) [inline code/game/gamemodes/meteor/meteors.dm]: Won't radiate through walls?

Comment (Alberyk) [inline code/game/gamemodes/meteor/meteors.dm]: Yes it will, but ,take in mind that meteor effect also happens with an explosion.

**Comment by Alberyk** [comment]
Changes done.

**Pull request merged**